ui.apps/src/main/content/jcr_root/apps/etoolbox-backpack/admin/console/components/openprompt/clientlibs/js/upload.js

```diff
@@ -24,30 +24,46 @@ $(function () {
     $(window).adaptTo('foundation-registry').register('foundation.form.response.ui.success', {
         name: 'backpack.prompt.open',
         handler: function (form, config, data, textStatus, xhr) {
-            function successPopup(ui, dataJson) {
-                ui.prompt(config.title, config.message, 'success', [{
+            function successPopup(ui, result, message, dataJson) {
+                ui.prompt(result.title, message, result.status, [{
                     text: Granite.I18n.get('Done'),
                     handler: function () {
-                        window.location.href = URITemplate.expand(config.redirect, dataJson);
+                        window.location.href = URITemplate.expand(result.redirect, dataJson);
                     }
                 }, {
                     text: Granite.I18n.get('Open'),
                     primary: true,
                     handler: function () {
-                        open(URITemplate.expand(config.open, dataJson), true);
-                        window.location.href = URITemplate.expand(config.redirect, dataJson);
+                        open(URITemplate.expand(result.open, dataJson), true);
+                        window.location.href = URITemplate.expand(result.redirect, dataJson);
                     }
                 }]);
             }
 
-            var ui = $(window).adaptTo('foundation-ui'),
-                dataJson = JSON.parse(data);
-
-            successPopup(ui, dataJson);
+            var ui = $(window).adaptTo('foundation-ui');
+            const resultConfig = Object.assign(config, {'status': textStatus}, data.resultStatus);
 
+            successPopup(ui, resultConfig, extractMessage(resultConfig), data);
         }
     });
 
+    function extractMessage(result) {
+        if (result.status === 'warning') {
+            let $wrapper = createEl('div', 'etoolbox-backpack-prompt_wrapper');
+            createEl('div').text(result.message).appendTo($wrapper);
+            createEl('div', 'etoolbox-backpack-prompt_wrapper-paths').html(createEl('pre').text(result.brokenPaths.join('\n'))).appendTo($wrapper);
+            return createEl('div').html($wrapper).html();
+        } else {
+            return result.message;
+        }
+    }
+
+    function createEl(name, className) {
+        const el = document.createElement(name);
```

Comment:
Just as a note, if you are going to use a Jquery wrapper and not a vanilla DOM element, there is no need to create a vanilla DOM element. Because eventually it will be the wrapper anyway. So you could go with one line: `return $('<div></div>').addClass(className);`